src/audio/wav64.c

```diff
@@ -165,7 +182,7 @@ void raw_waveform_read(samplebuffer_t *sbuf, int base_rom_addr, int wpos, int wl
 static void waveform_read(void *ctx, samplebuffer_t *sbuf, int wpos, int wlen, bool seeking) {
 	wav64_t *wav = (wav64_t*)ctx;
 	int bps = (wav->wave.bits == 8 ? 0 : 1) + (wav->wave.channels == 2 ? 1 : 0);
-	raw_waveform_read(sbuf, wav->rom_addr, wpos, wlen, bps);
+	raw_waveform_read(sbuf, wav->currentFile, wpos, wlen, bps);
```

Comment:
You should check `seeking` before calling this function. When `seeking` is `false`, the mixer guarantee that the requested read if subsequent to the previous one, so you could just call `read` without any seek. Instead, if `seeking` is true, you should really first `lseek` to `wpos << bps` and then call `read()`. Basically when seeking is false, `wpos` is just informative but it's not really required.